org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java

```diff
@@ -11,32 +11,17 @@
  *******************************************************************************/
 package org.eclipse.scanning.points;
 
-import java.io.File;
-import java.io.IOException;
 import java.util.Iterator;
-import java.util.Properties;
 
-import org.eclipse.core.runtime.FileLocator;
-import org.eclipse.core.runtime.Platform;
-import org.osgi.framework.Bundle;
-import org.osgi.framework.wiring.BundleWiring;
-import org.python.core.Py;
-import org.python.core.PyNone;
+import org.eclipse.scanning.jython.JythonObjectFactory;
 import org.python.core.PyObject;
-import org.python.core.PyString;
-import org.python.core.PySystemState;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /**
  * Based on JythonObjectFactory
  * See: http://www.jython.org/jythonbook/en/1.0/JythonAndJavaIntegration.html#more-efficient-version-of-loosely-coupled-object-factory
  */
 public class ScanPointGeneratorFactory {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1118)